modules/social_features/social_post/src/Hooks/SocialPostFormHooks.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Drupal\social_post\Hooks;
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\Core\Session\AccountProxyInterface;
+use Drupal\Core\StringTranslation\TranslatableMarkup;
+use Drupal\hux\Attribute\Alter;
+use Drupal\social_group\CurrentGroupService;
+use Drupal\social_post\Service\SocialPostHelperInterface;
+
+
+final class SocialPostFormHooks {
+
+  private SocialPostHelperInterface $socialPostHelper;
+  private AccountProxyInterface $currentUser;
+
+  public function __construct(
+    SocialPostHelperInterface $social_post_helper,
+    AccountProxyInterface $current_user,
+  ) {
+    $this->socialPostHelper = $social_post_helper;
+    $this->currentUser = $current_user;
+  }
+
+  /**
+   * From alter hook: replacement of social_post_form_post_form_alter.
+   *
+   * The method definition contains all standard parameters defined by the
+   * definition of the hook.
+   *
+   * @param array $form
+   * @param FormStateInterface $form_state
+   *
+   * @return array
+   */
+  #[Alter('form_post_form')]
+  public function formPostFormAlter(array &$form, FormStateInterface $form_state): void {
+    $content = $this->socialPostHelper->buildCurrentUserImage();
```

Comment:
If we want to respect Clean Code principles, this method should be split into something like
```
$form = $this->buildUserImage($form);
$form = $this->setCaption($form);
$form =$this->setTitleAndPlaceholder($form);
```
Thoughts? Suggestion on how far CCP should be applied?

Reply:
Splitting it up makes it easier to read and test.

It would be interesting to see the change in cyclomatic complexity too, I think it drops significantly.

Reply:
I think this one is always hard and eventually boils down to "How can we make this easiest to understand". I think for me "setFormTitleAndPlaceholder" is more complicated than having the two assignment statements repeated a few times. At other places I'm already dealing with the `$form` array, so it's weird that a part of the code then can't deal with the `$form` array anymore but delegates it. That feels like it mixes multiple levels of abstraction. Usually in this case I point developers to https://kentcdodds.com/aha